src/coreclr/src/utilcode/pedecoder.cpp

```diff
@@ -1770,20 +1770,31 @@ void PEDecoder::LayoutILOnly(void *base, BOOL allowFullPE) const
                            PAGE_READONLY, &oldProtection))
         ThrowLastError();
 
-    // Finally, apply proper protection to copied sections
-    section = sectionStart;
-    while (section < sectionEnd)
+    // Finally, apply proper protection to copied sections    
+    for (section = sectionStart; section < sectionEnd; section++)
     {
         // Add appropriate page protection.
-        if ((section->Characteristics & VAL32(IMAGE_SCN_MEM_WRITE)) == 0)
+#if defined(CROSSGEN_COMPILE) || defined(TARGET_UNIX)
+        if (section->Characteristics & IMAGE_SCN_MEM_WRITE)
+            continue;
+
+        DWORD newProtection = PAGE_READONLY;
```

Comment:
Isn't `DWORD` defined in `windows.h`? Why do we use it for xplat code?

Reply:
Consuming function is `BOOL ClrVirtualProtect(LPVOID lpAddress, SIZE_T dwSize, DWORD flNewProtect, PDWORD lpflOldProtect)`

In many windows style APIs are in use historically and PAL bridges to posix, if needed, or whatever.